samza-core/src/test/java/org/apache/samza/clustermanager/MockClusterResourceManager.java

```diff
@@ -102,12 +102,17 @@ public void launchStreamProcessor(SamzaResource resource, CommandBuilder builder
 
   @Override
   public void stopStreamProcessor(SamzaResource resource) {
-    SamzaResourceStatus status = new SamzaResourceStatus(resource.getContainerId(), "diagnostics", SamzaResourceStatus.PREEMPTED);
+    stopStreamProcessor(resource, SamzaResourceStatus.PREEMPTED);
+  }
+
+  void stopStreamProcessor(SamzaResource resource, int exitCode) {
```

Comment:
Is this required only for testing?

Reply:
Maybe @visible for testing or 
// package private for testing?

Reply:
this class is a Mock itself but sure Ill add it